app/services/doc_auth/lexis_nexis/responses/lexis_nexis_response.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module DocAuth
+  module LexisNexis
+    module Responses
+      class LexisNexisResponse < DocAuth::Response
+        PII_DETAILS = %w[
+          Age
+          DocIssuerCode
+          DocIssuerName
+          DocumentName
+          DOB_Day
+          DOB_Month
+          DOB_Year
+          FullName
+          Sex
+          Fields_Address
+          Fields_AddressLine1
+          Fields_AddressLine2
+          Fields_City
+          Fields_State
+          Fields_PostalCode
+          Fields_Height
+        ].freeze
+        attr_reader :http_response
+
+        def initialize(http_response)
+          @http_response = http_response
+          super(
+            success: successful_result?,
+            errors: error_messages,
+            extra: extra_attributes,
+            pii_from_doc: pii_from_doc,
+          )
+        end
+
+        def successful_result?
+          raise NotImplementedError
+        end
+
+        def error_messages
+          raise NotImplementedError
+        end
+
+        def extra_attributes
+          raise NotImplementedError
+        end
+
+        def pii_from_doc
+          raise NotImplementedError
+        end
+
+        private
+
+        def detail_groups
+          raise NotImplementedError
+        end
+
+        def parsed_response_body
+          @parsed_response_body ||= JSON.parse(http_response.body).with_indifferent_access
+        end
+
+        def status
+          @status ||= parsed_response_body.dig(:Status)
+        end
+
+        def conversation_id
+          @conversation_id ||= status.dig(:ConversationId)
+        end
+
+        def reference
+          @reference ||= status.dig(:Reference)
+        end
+
+        def transaction_status
+          @transaction_status ||= status.dig(:TransactionStatus)
+        end
+
+        def products
+          @products ||= begin
+            product_list = {}
+            parsed_response_body.dig(:Products).each do |product|
+              extract_details(product)
+              product_list[product[:ProductType]] = product
+            end
+            product_list.with_indifferent_access
+          end
```

Comment:
This is perhaps an over-complicated approach, but will help on the overall score on complexity, by removing the `begin` block:
```suggestion
          @products ||=
            parsed_response_body.dig(:Products).each_with_object({}) do |product, product_lis|
              extract_details(product)
              product_list[product[:ProductType]] = product
            end.with_indifferent_access
```

Reply:
I'll take it -- thanks!